annotation-processor/src/main/java/org/hibernate/validator/ap/classchecks/MethodOverrideCheck.java

```diff
@@ -0,0 +1,274 @@
+/*
+ * Hibernate Validator, declare and validate application constraints
+ *
+ * License: Apache License, Version 2.0
+ * See the license.txt file in the root directory or <http://www.apache.org/licenses/LICENSE-2.0>.
+ */
+package org.hibernate.validator.ap.classchecks;
+
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.TypeMirror;
+import javax.lang.model.util.Elements;
+import javax.lang.model.util.Types;
+
+import org.hibernate.validator.ap.checks.ConstraintCheckError;
+import org.hibernate.validator.ap.util.CollectionHelper;
+
+/**
+ * Abstract base class for {@link ClassCheck} implementations of overridden method checks.
+ *
+ * @author Marko Bekhta
+ */
+public abstract class MethodOverrideCheck extends AbstractClassCheck {
+
+	private static final String JAVA_LANG_OBJECT = "java.lang.Object";
+
+	private static final String JAVAX_VALIDATION_CONSTRAINT = "javax.validation.Constraint";
+
+	private final Elements elementUtils;
+
+	private final Types typeUtils;
+
+	public MethodOverrideCheck(Elements elementUtils, Types typeUtils) {
+		this.elementUtils = elementUtils;
+		this.typeUtils = typeUtils;
+	}
+
+	@Override
+	public Collection<ConstraintCheckError> checkMethod(ExecutableElement currentMethod) {
+		if ( !needToPerformAnyChecks( currentMethod ) ) {
+			return Collections.emptySet();
+		}
+		TypeElement currentTypeElement = getEnclosingTypeElement( currentMethod );
+
+		// find if there's a method that was overridden by a current one.
+		List<ExecutableElement> overridden = findAllOverriddenElements( currentTypeElement, currentMethod );
+		if ( overridden.isEmpty() ) {
+			return Collections.emptySet();
+		}
+
+		// if there's more than one overridden method we need to make sure that all of them match
+		Set<ConstraintCheckError> errors = CollectionHelper.newHashSet();
+		for ( ExecutableElement firstExecutable : overridden ) {
```

Comment:
@gsmet thanks for the review. And no worries about the delay, I completely understand it - this one is not as easy as adding translation for messages :) 

As for what you wrote it seems to make sense to me... But I need to think about it a bit before I get back to you with a good answer/solution. 

Also just as information about what I was trying to do by this part of code when I wrote it - it supposed to be only responsible for finding issues in parallel declarations. As the check in hierarchy tree is done in the loop on line 77. And also reading the comment that I wrote - `we need to make sure that all of them match` - it seems that it's doing exactly that (when I was implementing this I thought that if there are two "parallel methods" then their constraints should match, but obviously there can be methods in this list that can have different constraints) - but actually it might not be the right thing to check....

So this is what I'll do next - I'll prepare a few more different test cases and then I'll see how this parallel check can be corrected.